internal/app/db/connection/postgresql.go

```diff
@@ -0,0 +1,18 @@
+package connection
+
+import (
+	"database/sql"
+	"log"
+
+	_ "github.com/lib/pq"
+)
+
+func NewDBPool(dsn string) (*sql.DB, error) {
+	pool, err := sql.Open("postgres", dsn)
+	if err != nil {
+		log.Fatal("unable to use data source name", err)
```

Comment:
```suggestion
		return nil, err
```
И уже наверху решать - падать по Fatal или делать fallback на другое хранилище.